### PR TITLE
migrate BatchNorm kernels to sycl free function (#179)

### PR DESCRIPTION
**Pull request by jiafuzha**
torch-xpu-ops regression │  pytest third_party/torch-xpu-ops/test/regressions/test_batch_norm.py -v          │ 7 passed  
torch-xpu-ops XPU OpInfo │  pytest third_party/torch-xpu-ops/test/xpu/test_ops_xpu.py -v -k "batch_norm"     │ 86 passed, 5 skipped, 4 xfailed 
torch-xpu-ops XPU decomp │  pytest third_party/torch-xpu-ops/test/xpu/test_decomp_xpu.py -v -k "batch_norm"  │ 23 passed, 12 skipped
PyTorch core decomp      │  pytest test/test_decomp.py -v -k "batch_norm"                                    │ 22 passed, 9 skipped
PyTorch core OpInfo      │  pytest test/test_ops.py -v -k "batch_norm"                                       │ 69 passed, 48 skipped, 7 xfailed

### CODE REVIEW

**Comment by tye1** [comment]
@jiafuzha  pls. fix the code conflicts and CI issues
